src/Config/RabbitMQExtensionConfigProvider.cs

```diff
@@ -51,27 +50,22 @@ public void Initialize(ExtensionConfigContext context)
             rule.AddOpenConverter<OpenType.Poco, byte[]>(typeof(PocoToBytesConverter<>));
 
             var triggerRule = context.AddBindingRule<RabbitMQTriggerAttribute>();
-            triggerRule.BindToTrigger<BasicDeliverEventArgs>(new RabbitMQTriggerAttributeBindingProvider(
+
+            // More details about why the BindToTrigger was chosen instead of BindToTrigger<BasicDeliverEventArgs> detailed here https://github.com/Azure/azure-functions-rabbitmq-extension/issues/110
+            triggerRule.BindToTrigger(new RabbitMQTriggerAttributeBindingProvider(
                     _nameResolver,
                     this,
                     _logger,
                     _options,
                     _configuration));
-
-            // Converts BasicDeliverEventArgs to string so user can extract received message.
-            triggerRule.AddConverter<BasicDeliverEventArgs, string>(args => Encoding.UTF8.GetString(args.Body))
-                .AddConverter<BasicDeliverEventArgs, DirectInvokeString>((args) => new DirectInvokeString(null));
-
-            // Convert BasicDeliverEventArgs --> string-- > JSON-- > POCO
-            triggerRule.AddOpenConverter<BasicDeliverEventArgs, OpenType.Poco>(typeof(BasicDeliverEventArgsToPocoConverter<>), _logger);
         }
 
         public void ValidateBinding(RabbitMQAttribute attribute, Type type)
         {
             string connectionString = Utility.FirstOrDefault(attribute.ConnectionStringSetting, _options.Value.ConnectionString);
             string hostName = Utility.FirstOrDefault(attribute.HostName, _options.Value.HostName) ?? Constants.LocalHost;
             _logger.LogInformation("Setting hostName to localhost since it was not specified");
-            string queueName = Utility.FirstOrDefault(attribute.QueueName, _options.Value.QueueName);
+            _ = Utility.FirstOrDefault(attribute.QueueName, _options.Value.QueueName);

```

Comment:
Can this code be removed completely? Looks like the value isn't being used.

Reply:
removed